definitions/checks/yum_exclude.rb

```diff
@@ -0,0 +1,23 @@
+class Checks::YumExclude < ForemanMaintain::Check
+  metadata do
+    label :check_yum_exclude_list
+    description 'Check if yum exclude list is configured'
+    tags :pre_upgrade
+  end
+
+  def run
+    assert(exclude_set?, 'The /etc/yum.conf has exclude list configured as below,'\
+          "\n  #{yum_exclude}"\
+          "\nUnset this as it can cause yum update or upgrade failures !")
+  end
+
+  def exclude_set?
+    yum_exclude == 'exclude=' || yum_exclude == 'exclude ='
+  end
+
+  def yum_exclude
+    execute!('grep -w exclude /etc/yum.conf')
+  rescue ForemanMaintain::Error::ExecutionError
+    'exclude='
```

Comment:
This feels quite hacky, could we return empty string here?
Perhaps we could call the method `grep_yum_exclude` to see better what it does.

Reply:
I changed the method name and now using execute_with_status which returns empty string when exclude is not set.